cmd/soroban-cli/src/commands/keys/add.rs

```diff
@@ -31,9 +43,32 @@ pub struct Cmd {
 impl Cmd {
     pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
         let print = Print::new(global_args.quiet);
-        let secret = self.secrets.read_secret()?;
+        let secret = self.read_secret(&print)?;
         let path = self.config_locator.write_identity(&self.name, &secret)?;
         print.checkln(format!("Key saved with alias {:?} in {path:?}", self.name));
         Ok(())
     }
+
+    fn read_secret(&self, print: &Print) -> Result<Secret, Error> {
+        if let Ok(secret_key) = std::env::var("SOROBAN_SECRET_KEY") {
+            return Ok(Secret::SecretKey { secret_key })
+        } else if self.secrets.secure_store {
+            let prompt = "Type a 12 or 24 word seed phrase:";
+            let secret_key = read_password(print, prompt)?;
+
+            let seed_phrase: SeedPhrase = secret_key.parse()?;
```

Comment:
At the moment I only added support for a user to add a new **seed phrase** to secure storage. I'm happy to make a change so that this supports adding a private key directly as well, but wanted to get some more eyes on this to see if folks thought it was worth the effort.

Reply:
I think it's always a mistake to use a private key directly when storing keys. It has no recovery method and almost all wallets to my knowledge use a seed phrase so if users would be importing it wouldn't be a Secret Key.